### DISCUSSION TITLE
Crash on startup (version `GLIBCXX_3.4.29' not found , tags: linux, liblibveldrid-spirv)

### DISCUSSION THREAD

**Discussion by McModknower** [ppy/osu#22976] [Q&A] [answered]
### Type

Crash to desktop

### Bug description

I tried to start the game and it just crashed.

I am using linux mint with $(uname -a) = `Linux teddy 5.4.0-144-generic #161-Ubuntu SMP Fri Feb 3 14:49:04 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`

The line
```
/usr/lib/x86_64-linux-gnu/libstdc++.so.6: error: version lookup error: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_osu.ApKf2Cz6/usr/bin/libveldrid-spirv.so) (fatal)
```
looks to me like it is a library version mismatch, but i have to go afk for some time now. If this isn't fixed when i come back (in 30min - 2 hours), i will some more things to narrow it down.

### Screenshots or videos

N/A

### Version

2023.326.1

### Logs

No logs were written, but starting it as `LD_DEBUG=reloc ./osu.AppImage` via zsh resulted in the attached error (i truncated it a bit at the top, if you need more, tell me).
[osu-error.log](https://github.com/ppy/osu/files/11072581/osu-error.log)

**Comment by bdach** [comment, marked as answer]
Sorta duplicate of #22967 but sure.

I mean it says it all on the tin really; your system's glibc version is too low to be compatible with the binaries we're shipping. I'm not sure we'll be fixing this - right now seems like your best chance is an upgrade.

What version of linux mint iis this?

**Reply by McModknower** [reply]
Ah, i didn't see that other issue.

I'm currently using linux mint 20.2.

Updating to linux mint 21.* is currently not an option, as i need the ubuntu focal base for a university project.

I just tested the flatpak version (the test build https://github.com/flathub/sh.ppy.osu/pull/99#issuecomment-1484171702) and it works. So i will just use the flatpak version until i upgrade my system.
That might also be a useful way for others who have this problem.

Also just a quick question: I thought AppImages contained all libraries that they use. Is that just not the case for libstdc++?

**Reply by bdach** [reply]
You have to go out of your way quite far to compensate for glibc version incompatibilities when shipping appimages. See; https://docs.appimage.org/reference/best-practices.html#binaries-compiled-on-old-enough-base-system

**Reply by McModknower** [reply]
Ok, thanks for the info.
Should i close this discussion now, since all my questions are answered?

**Reply by Thomzzz** [reply]
I had the same problem with mint 20.2.
Just now did a fresh install with mint 21.1 and I can confirm that this solves the issue with glibc.
